src/engine/tests/Test_SignConstraint.h

```diff
@@ -931,4 +931,63 @@ class SignConstraintTestSuite : public CxxTest::TestSuite
         TS_ASSERT_EQUALS( originalSign.serializeToString(),
                           recoveredSign.serializeToString() );
     }
+
+      void test_polarity()
+    {
+        unsigned b = 1;
+        unsigned f = 4;
+
+        // b in [1, 2], polarity should be 1, and direction should be PHASE_POSITIVE
+        {
+            SignConstraint sign( b, f );
+            sign.notifyLowerBound( b, 1 );
+            sign.notifyUpperBound( b, 2 );
+            TS_ASSERT( sign.computePolarity() == 1 );
+
+            sign.updateDirection();
+            TS_ASSERT( sign.getDirection() == SignConstraint::PHASE_POSITIVE );
+        }
+        // b in [-2, 0], polarity should be -1, and direction should be PHASE_NEGATIVE
+        {
+            SignConstraint sign( b, f );
+            sign.notifyLowerBound( b, -2 );
+            sign.notifyUpperBound( b, 0 );
+            TS_ASSERT( sign.computePolarity() == -1 );
+
+            sign.updateDirection();
+            TS_ASSERT( sign.getDirection() == SignConstraint::PHASE_NEGATIVE );
```

Comment:
@anwu1219 @guykatzz I think this is incorrect because of the upper bound 0. Only if strictly smaller than 0 then the phase is NEGATIVE. Agree?

Reply:
That makes sense. I pushed a fix.